src/main/scala/io/iohk/ethereum/ledger/BlockPreparator.scala

```diff
@@ -376,12 +376,11 @@ class BlockPreparator(
     }
   }
 
-  def prepareBlock(block: Block): BlockPreparationResult = {
-    val parentStateRoot = blockchain.getBlockHeaderByHash(block.header.parentHash).map(_.stateRoot)
+  def prepareBlock(block: Block, parent: BlockHeader): PreparedBlock = {
```

Comment:
I think this addresses https://jira.iohk.io/browse/ETCM-329 but only if the mocked miner doesn't use any txs, if you:
1. Mine 30 blocks, i.e. with numbers 1-30
2. Set to mine 30 blocks having block 1 as it's parent
3. Send a tx before step 2. completes

It will fail as in ETCM-329... we had solved that in the other project by passing here the `parentState: InMemoryWorldState` instead of having to fetch it here